organization/_posts/2020/2020-09-30-packaging.md

```diff
@@ -0,0 +1,115 @@
+    - Henry: yes, key thing is the "Do" on testing with lowest version you want to support
+    - Example of a bad policy in CMake 3.9:
+        - breaks IPO
+    - Thus the recommendation on don't setting the min version low without a range
+    - Attila: big problem is updating min version through a project's file
+    - Henry: one recommendation here (though CMake didn't do this themselves) is not to set minimum required on anything other than `CMakeLists.txt`.
+- Minimum versions by OS (e.g. "out-the-box" on CentOS etc), vs CMake features (what it can do)
+- Installing CMake now very easy across all platforms (inc. GitLab/Hub CI)
+    - See the installing page on the website
+- Modern cmake has better CLI, so can run underlying configure/build/test/install directly through the `cmake` command
+- Key CMake convention: property initialization
+    - A property `MY_PROPERTY` initialized by `CMAKE_MY_PROPERTY`
+- Key tips for packaging:
+    - Use targets
+    - Don't write a "FindMyPackage", write "MyPackageConfig"
+    - Don't hardcode system/compiler/flags in targets
+- Downloading dependencies
+    - classic: ExternalProject (build time)
+    - new: FetchContent, (configure time)
+        - Like submodules, much easier integration, direct use of targets provided by subproject.
+- Can also integrate with Conan/Conan.io
+    - requires conan (but can bootstrap it in your cmake script)
+- Imported targets
+    - A target that comes from outside the project
+    - Cmake 3.11 now makes these much easier to define much like all other targets
+- CUDA a first class language from 3.8/9, more development since then, including
+    - C++ standards
+    - .cu files automatically handled with nvcc
+- Attila: should things like CMAKE_CXX/CUDA_STANDARD be cached (so can override)?
+    - Henry: yes, and also consider using compile features
+- Chris: Is it possible to have a subproject that uses a language that the top level project doesn't?
+    - Henry: it should be, but would need to check
+    - Chris: have had this fail up to 3.14
+    - To be checked!
+- Major updates to CUDA support in 3.17/18
+    - architectures, clang, shared cudart, FindCUDAToolkit
+- Extra tools and integrations
+    - IPO/PIC
+    - ccache
+    - clang-tidy, clang tooling
+    - Work well with GitHub Actions (and ci)
+- Also a cmake-format tool now
+    - Just like black/clang-format for CMake
+    - Builtin GitHub Action support
+- Use try_compile/try_run/WriteCompilerDetectionHeader to check/addpt to different compilers
+- Python and CMake
+    - New "FindPython" module is recommended over separate Interp/Libs modules
+    - Best since 3.15, PyPy needs 3.18
+    - Attila: Quite slow in ATLAS due to number of calls
+    - Henry: due to behaviour of FindPython in supporting multiple versions, so trying to refind Python each call. Is a variable that can be set to disable that. Looks like this is new in 3.18. Can vendor this in projects if you can't require 3.18.
```

Comment:
Add note that this is activated with `Python_ARTIFACTS_INTERACTIVE=TRUE` as it's not that obvious from the documentation that this will have this effect of speeding things up to fix the problem Attla reported.